tests/lib/rules/no-unsafe-optional-chaining.js

```diff
@@ -0,0 +1,231 @@
+/**
+ * @fileoverview Tests for no-unsafe-optional-chaining rule.
+ * @author Yeon JuAn
+ */
+
+"use strict";
+
+const rule = require("../../../lib/rules/no-unsafe-optional-chaining");
+
+const { RuleTester } = require("../../../lib/rule-tester");
+
+const parserOptions = {
+    ecmaVersion: 2021,
+    sourceType: "module"
+};
+
+const ruleTester = new RuleTester({ parserOptions });
+
+ruleTester.run("no-unsafe-optional-chaining", rule, {
+    valid: [
+        "var foo;",
+        "class Foo {}",
+        "!!obj?.foo",
+        "obj?.foo();",
+        "obj?.foo?.();",
+        "(obj?.foo ?? bar)();",
+        "(obj?.foo)?.()",
+        "(obj?.foo ?? bar?.baz)?.()",
+        "(obj.foo)?.();",
+        "obj?.foo.bar;",
+        "obj?.foo?.bar;",
+        "(obj?.foo)?.bar;",
+        "(obj?.foo)?.bar.baz;",
+        "(obj?.foo)?.().bar",
+        "(obj?.foo ?? bar).baz;",
+        "(obj?.foo ?? val)`template`",
+        "new (obj?.foo ?? val)()",
+        "new bar();",
+        "obj?.foo?.()();",
+        "const {foo} = obj?.baz || {};",
+        "bar(...obj?.foo ?? []);",
+        "var bar = {...foo?.bar};",
+        "foo?.bar in {};",
```

Comment:
Can we add one valid test to make sure that the relation operator check applies to `in` and `instanceof` only, for example:

```js
"foo?.bar < foo?.baz;"
```